### ISSUE TITLE
A way to set format option for getContents

### ISSUE THREAD

**Issue by rdpeake** [tinymce/tinymce-angular#120]
**Is your feature request related to a problem? Please describe.**
In ui-tinymce there was an option to set the 'format' value used with get contents. we have migrated our website to use later versions of angular and would like to use 'text' format on some instances of tinymce

 **Describe the solution you'd like**
An option to pass in a format that is used with 'getContents' such that a text format of the editors content is extracted

 **Describe alternatives you've considered**
The only option i can think to get this without direct support (with formControlName) is to not use formControlName but manually convert the value using contentChanged callbacks, but not sure how to get access to the editor instance inside the component (maybe ViewChild) - any advise on this would be nice as well.

**Comment by SimonFc** [comment]
I'm not familiar with how AngularJS and the ui-tinymce wrapper does things but with Angular 2+ and the tinymce-angular wrapper I don't think you would want the format of the value that binds TinyMCE with Angular to be anything other that HTML.

`tinymce-angular -> MyAwesomeState` would work with `{ format: 'text' }` but the flow the other way would be strange since you would lose all the structure.

All [events handlers](https://github.com/tinymce/tinymce-angular#event-binding) are called with the editor instance so if you provide an handler to the onInit-event (for example) you can get access to the editor
```
<editor (onInit)="handleInit($event)"></editor>
/* ... */
public handleInit({ event, editor }) {
  console.log(editor.getContent({ format: 'text' }));
}
```

You can use ViewChild as well since we do expose the editor-instance for scenarios like these: [example](https://github.com/tinymce/tinymce-angular/blob/master/stories/viewquery/viewquery.component.ts)

Does this solve your issue?

**Comment by rdpeake** [comment]
I resolved my issue locally by using patch-package to make a minor changed to my local copy:

added:
```js
@Input() public format='html'
```
and modified https://github.com/tinymce/tinymce-angular/blob/53e523691bace7f0ca23d0f6f077b9b028811236/tinymce-angular-component/src/main/ts/editor/editor.component.ts#L163 to

```js
   editor.on('change keyup undo redo', () => this.ngZone.run(() => this.onChangeCallback(editor.getContent({format: this.format))));
```

In the default case it functions the same, whilst allowing you to get 'raw', 'html', 'text', or 'tree' as supported by tinymce's get content. We use the rich text editor to markup plain text, but rather than replacing all html using plugins, just use the raw text feature and limit the UI (given we plan to also use a full html editor as well, using 2 different editors seemed needlessly complex)

**Comment by rdpeake** [comment]
I'm getting plain text from tinymce, and i'm setting the initial value as plain text. I can, whilst editing the value add graphics and such to make editing things really intuitive (via plugins), whilst still storing plain text from tinymce. Just seems odd to not include a feature the base editor has built in given it doesn't break anything?

**Comment by rdpeake** [comment]
Also given this http://archive.tinymce.com/wiki.php/API4:method.tinymce.Editor.setContent, and it's comment about the 'bbcode plugin' there may actually be times when exposing a format and using it with both the set, and get contents methods of tinymce would actually be required for propper functionality - even if my use of text seems incorrect.

**Comment by SimonFc** [comment]
You're absolutely right about the BBCode example, a different format is required there. And if we set the format setting for [setContent](https://github.com/tinymce/tinymce-angular/blob/master/tinymce-angular-component/src/main/ts/editor/editor.component.ts#L73) as well it'll work both ways as expected, sorry I overlooked that.

**Comment by SimonFc** [comment]
This was fixed in 3.4.0 where a new `outputFormat` property was added which can take the value of either `html` or `text`. The `bbcode` format doesn't actually do anything so that example for the setContent function has been fixed.
